lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -408,11 +408,11 @@ class SetVariantSelectionUndoableCommand : public Ufe::UndoableCommand
         , _oldSelection(_varSet.GetVariantSelection())
         , _newSelection(itemPath[2])
     {
-        const bool allowStronger = true;
-        MayaUsd::ufe::applyCommandRestriction(
-            prim,
-            "set variant set " + _varSet.GetName() + " to variant " + _newSelection + " on",
-            allowStronger);
+        std::string errMsg;
+        if (!MayaUsd::ufe::isPrimMetadataEditAllowed(
+                prim, SdfFieldKeys->VariantSelection, TfToken(_varSet.GetName()), &errMsg)) {
+            throw std::runtime_error(errMsg.c_str());
+        }
     }
```

Comment:
In the command we call the new check for metadata editing instead of general prim-level command restriction.